CondFormats/CTPPSReadoutObjects/plugins/CTPPSBeamParametersESSource.cc

```diff
@@ -0,0 +1,238 @@
+/****************************************************************************
+ *
+ *  CondFormats/CTPPSReadoutObjects/plugins/CTPPSBeamParametersESSource.cc
+ *
+ *  Description :  - Loads CTPPSBeamParameters from the CTPPSBeamParametersESSource_cfi.py
+ *                   config file.
+ *                 - Currently, one single set of beam parameters is provided. Just to be
+ *                   ready in time for 10_4_0 and allow simple tests.
+ *                 - To be further developed to provide multiple sets of parameters
+ *                 - Needed while CTPPSBeamParameters is not available in database
+ *
+ *
+ * Author:
+ * Wagner Carvalho wcarvalh@cern.ch
+ *
+ ****************************************************************************/
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/SourceFactory.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+#include "FWCore/Framework/interface/ESProducts.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSBeamParameters.h"
+#include "CondFormats/DataRecord/interface/CTPPSBeamParametersRcd.h"
+
+#include <memory>
+
+//----------------------------------------------------------------------------------------------------
+
+using namespace std;
+
+/**
+ * \brief Loads CTPPSBeamParameters from a config file.
+ **/
+
+class CTPPSBeamParametersESSource: public edm::ESProducer, public edm::EventSetupRecordIntervalFinder
```

Comment:
I see that this is not the first example, however CTPPS is the only case with ESProducers defined in CondFormats.
Isn't there some rule similar to the DataFormats/ that the data format packages should not contain producers

Reply:
Yes, we have the same policy for ESProducers as for Producers.